modules/cluster-cloud-controller-config-osp.adoc

```diff
@@ -0,0 +1,296 @@
+// Module included in the following assemblies:
+//
+// * installing/installing_openstack/installing-openstack-cloud-config-reference.adoc
+
+:_content-type: REFERENCE
+[id="cluster-cloud-controller-config_{context}"]
+= The {rh-openstack} Cloud Controller Manager (CCM) config file
+
+A {rh-openstack-first} CCM config file, called `cloud.conf` by convention, comprises several options in an INI format.
```

Comment:
@stephenfin @MaysaMacedo I think it is not clear, or at least for me, what config file is detailed here. AFAIK, user should modify only the configmap `cloud-provider-config` in the `openshift-config namespace`. Don't you think that should be mention at the top of the doc?

On the other hand, if the intention is to detail the content of the `cloud-conf` configMap in the `openshift-cloud-controller-manager`, all the parameters should be included in the informative section, as the user should never modify that configMap, right?

Reply:
@rlobillo The config detailes here is the user managed config-map present on openshift-config namespace and it's ok if mention that here. I think we should document only the sections that will have an effect if modified by the user, which is what seems to be targeted in the PR

Reply:
Check out the latest iteration of this PR and LMK.

Reply:
Hello @maxwelldb I think it's the other way around: The file that the user should modify is the configmap `cloud-provider-config` in the `openshift-config` namespace, and according to @MaysaMacedo and I agree with her, that should be one mentioned on the top of the doc.

Reply:
@rlobillo See https://github.com/openshift/openshift-docs/pull/53068/files#r1054406909

Reply:
Hello Max! No, that is not what I meant, sorry, probably I wasn't clear. Let me rephrase starting from the beginning:

1. If the user wants to change something on CCM, it should modify below configMap (`cloud-provider-config` on `openshift-config` namespace):

```
$ oc get configMap/cloud-provider-config --namespace openshift-config -o json | jq -r .data.config 
[Global]
secret-name = openstack-credentials
secret-namespace = kube-system
region = regionOne
ca-file = /etc/kubernetes/static-pod-resources/configmaps/cloud-config/ca-bundle.pem
[LoadBalancer]
lb-provider = amphora
lb-method = SOURCE_IP
create-monitor = True
monitor-delay = 10s
monitor-timeout = 10s
monitor-max-retries = 1
floating-network-id = 03c0b9b2-5994-4611-a7dd-a95df903c2f9
```

2. Then, according to what it is explained on https://github.com/openshift/openshift-docs/pull/52520, the OCP cluster is internally (transparent for the user) converting that new configuration to the configMap called cloud-conf on the `openshift-cloud-controller-manager` namespace:

```
$ oc get configMap/cloud-conf --namespace openshift-cloud-controller-manager -o json | jq -r '.data["cloud.conf"]'
[Global]
region      = regionOne
ca-file     = /etc/kubernetes/static-pod-resources/configmaps/cloud-config/ca-bundle.pem
use-clouds  = true
clouds-file = /etc/openstack/secret/clouds.yaml
cloud       = openstack

[LoadBalancer]
lb-provider         = amphora
lb-method           = SOURCE_IP
create-monitor      = True
monitor-delay       = 10s
monitor-timeout     = 10s
monitor-max-retries = 1
floating-network-id = 03c0b9b2-5994-4611-a7dd-a95df903c2f9
use-octavia         = true
``` 
So, my point is that we should mention at the top of the doc that **if a user want to change something on the cluster, he/she should modify the cloud-provider-config configMap in the openshift-config namespace** and not in the configMap on the openshift-cloud-controller-manager namespace.

Reply:
:+1: How do you feel about this change, then: https://github.com/openshift/openshift-docs/pull/53068/commits/d73f8fc00b66ea729f698bbc55446c7d1a1fc52b

@rlobillo

Reply:
that is great, thanks!